src/compiler/emitter.ts

```diff
@@ -2779,7 +2779,7 @@ namespace ts {
             if (isSimilarNode && currentSourceFile) {
                 pos = skipTrivia(currentSourceFile.text, pos);
             }
-            if (emitLeadingCommentsOfPosition && isSimilarNode && contextNode.pos !== startPos) {

```

Comment:
Was this formerly optional? Is there some other condition that it's supposed to be checking here?

Reply:
Not sure about other condition, `emitLeadingCommentsOfPosition` refers to the function

https://github.com/microsoft/TypeScript/blob/aa2756a5d7f5ed02e6c676e0fd8e4d1a5e0acd13/src/compiler/emitter.ts#L5195-L5201

I cannot find other places where it is destroyed. Maybe there are some reasons for this condition, however, if `emitLeadingCommentsOfPosition` can be `undefined`, we can check that `emitLeadingCommentsOfPosition` is `function`, or `!!emitLeadingCommentsOfPosition`.

Reply:
On the flip side, the correct change is probably to call `emitLeadingCommentsOfPosition`; but this may break some things that relied on it being incorrect. Do you mind trying?

(I assume the intended argument is `pos`, but maybe it's `startPos`?) @rbuckton  maybe you can enlighten us here.

Reply:
I've checked the emitter, and there is one case that I missed. If a binary expression used in condition statement, and function is used in condition body but not in binary expression. Do we need to check the parent of binary expression (`if, while, etc.`) and check the function reference in the body?

```ts
function a() {}
const b = 1;

if (a && b) { // foo wasn't used in binary expression
    a(); // however, foo was used in body
}
```

Reply:
@sandersn I've checked the emitter, and there is one case that I missed. If a binary expression used in condition statement, and function is used in condition body but not in binary expression. Do we need to check the parent of binary expression (`if, while, etc.`) and check the function reference in the body?

```ts
function a() {}
const b = 1;

if (a && b) { // a wasn't used in binary expression
    a(); // however, a was used in body
}
```
cc @RyanCavanaugh @rbuckton

Reply:
I talked to @rbuckton about this yesterday while we were looking at a bug in the emitter. Turns out that `emitLeadingCommentsOfPosition` (and friends) were all optional at one point, but now they're always defined. So the error is correct in this case -- they need to be removed.

Reply:
Oh, whoops, looks like you already did that. :sweat_smile:

Reply:
Oh, I didn't see this until after I merged the PR:

> Do we need to check the parent of binary expression (if, while, etc.) and check the function reference in the body?

Maybe yes? If it didn't show up in RWC and only showed up in our code as a correct error, we may not need to worry about it. Let's see what happens with the user code tests -- there should be a new build in about 20 minutes.